addons/binding/org.openhab.binding.gpio/src/main/java/org/openhab/binding/gpio/handler/GPIODigitalInputHandler.java

```diff
@@ -0,0 +1,125 @@
+package org.openhab.binding.gpio.handler;
+
+import static org.openhab.binding.gpio.GPIOBindingConstants.*;
+
+import java.math.BigDecimal;
+import java.util.Date;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jpigpio.Alert;
+import jpigpio.GPIO;
+import jpigpio.JPigpio;
+import jpigpio.PigpioException;
+
+public class GPIODigitalInputHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(GPIODigitalInputHandler.class);
+    private JPigpio jPigpio = null;
+    private GPIO gpio;
+    private Date lastChanged;
+
+    public GPIODigitalInputHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            updateState(getThing().getChannel(THING_TYPE_DIGITAL_INPUT_CHANNEL).getUID(), getValue());
+        } catch (PigpioException e) {
+            e.printStackTrace();
```

Comment:
Do not use `e.printStackTrace();` use real loggers and forward the exception to them instead